### DISCUSSION TITLE
[4.9] okd cannot create route object automatically through ingress object.

### DISCUSSION THREAD

**Discussion by hirsaeki-mki** [okd-project/okd#1309] [General]
I am trying to run a helm chart that has some Ingress objects from our ecosystem in okd. So I need to check this feature.

**Describe the bug**

The following steps were taken according to the description in the link(https://docs.okd.io/4.9/networking/routes/route-configuration.html#nw-ingress-creating-a-route-via-an-ingress_route-configuration).

  1. Create a new project and app.
  ```bash
  > export KUBECONFIG=bare-metal/auth/kubeconfig
  > oc new-project hello-openshift
  > oc new-app openshift/hello-openshift
  > nohup oc port-forward svc/hello-openshift 8080 &
  > curl localhost:8080
  Hello OpenShift!
  
  > pkill oc
  ``` 
  2. Create a Ingress Object
  ```bash
   > oc apply -f ingress-test.yaml
   ingress.networking.k8s.io/hello-openshift created   
   
   > oc get ingress hello-openshift -oyaml
   apiVersion: networking.k8s.io/v1
   kind: Ingress
   metadata:
     creationTimestamp: "2022-08-12T20:08:45Z"
     generation: 1
     name: hello-openshift
     namespace: hello-openshift
     resourceVersion: "154666"
     uid: e943e1ab-76d0-4112-8036-51d5fa38fa74
   spec:
     rules:
     - host: hello-openshift.okd.test
       http:
         paths:
         - backend:
             service:
               name: hello-openshift
               port:
                 number: 8080
           path: /
           pathType: Exact
  ```
  3. Check Route Object
  ```
  > oc get route
  No resources found in hello-openshift namespace.
  ```
**Version**

```
> oc version
Client Version: 4.9.0-0.okd-2022-02-12-140851
Server Version: 4.9.0-0.okd-2022-02-12-140851
Kubernetes Version: v1.22.1-1839+b93fd35dd03051-dirty
```
bare-metal installation via UPI

**How reproducible**

100%

**Log bundle**

Please tell me how to collect the necessary logs for this issue.

**Comment by vrutkovs** [comment]
Please attach (or upload to the public file sharing service) [must-gather](https://docs.okd.io/latest/support/gathering-cluster-data.html#support_gathering_data_gathering-cluster-data) archive

Also, is this reproducible on OKD 4.11?

**Comment by hirsaeki-mki** [comment]
The following is a must-gather of 4.9.
https://dropmefiles.com/k94S8
4.11 will be deployed now, so please wait a while.

**Comment by hirsaeki-mki** [comment]
The results were the same in the 4.11 environment
```
> oc version
Client Version: 4.11.0-0.okd-2022-07-29-154152
Kustomize Version: v4.5.4
Server Version: 4.11.0-0.okd-2022-07-29-154152
Kubernetes Version: v1.24.0-2364+954643139fe994-dirty
```
[must-gather-4.11.tar.gz](https://github.com/openshift/okd/files/9333033/must-gather-4.11.tar.gz)
.

**Comment by hirsaeki-mki** [comment]
Should this be returned to issues?

**Reply by vrutkovs** [reply]
No, we use issues for OKD specific bit we need to fix. If its not a misconfiguration we'll file an OCP bug - and if its OKD specific we'll return it to issues with a clear message which part needs to be fixed

**Comment by hirsaeki** [comment]
I changed the pathType from Exact to Prefix.
Route Object was created.

Does this mean that OKD does not support PathType=Exact?

**Reply by jrod-97** [reply]
Having the same issue. PathType=Exact does not create a route while Prefix works fine.
Any info if this is intended behavior?
